core/src/main/java/io/opencensus/tags/Internal.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import java.util.Iterator;
+
+/**
+ * Internal tagging utilities.
+ */
+@io.opencensus.common.Internal
+public final class Internal {
```

Comment:
Where do you use this class?

Reply:
Internal.getTags() is used by the stats implementation in MutableViewData, and by the tagging implementation in TaggerImpl and SerializationUtils.

Reply:
If it is used by the stats/tags implementation only why do we need it in the public API? Stats and Tags can get this from TagContextImpl.

Reply:
The implementation only accesses the internal state when the TagContext is a TagContextImpl.  It uses Internal.getTags() for any other subclass.  That way the TagContext's class only affects performance, not behavior.

Reply:
I don't think that the current design optimizes for that use case, it just ensures that the code behaves correctly on any input.  I think that having public methods that take arguments of type TagContext but only work on one subclass is confusing and not very user-friendly, even if it is unlikely that they will need to be called with different types.  This design choice also relates to the general argument against using `instanceof` in Java.